src/test/java/org/broadinstitute/consent/http/db/DataAccessRequestDAOTest.java

```diff
@@ -631,17 +631,17 @@ void testFindAllDatasetApprovalsByDar() {
     dataAccessRequestDAO.insertDARDatasetRelation(testDar1.getReferenceId(), dataset2.getDatasetId());
     dataAccessRequestDAO.insertDARDatasetRelation(testDar1.getReferenceId(), dataset3.getDatasetId());
 
-    Election e1 = createDataAccessElection(testDar1.getReferenceId(), dataset1.getDatasetId());
+    // Set up all elections for a single DAR on multiple datasets
+    Date now = new Date();
+    Election e1 = createDataAccessElection(testDar1.getReferenceId(), dataset1.getDatasetId(), now);
     Vote v1 = createFinalVote(dataset1.getCreateUserId(), e1.getElectionId());
 
-    Election e2 = createDataAccessElection(testDar1.getReferenceId(), dataset2.getDatasetId());
+    Election e2 = createDataAccessElection(testDar1.getReferenceId(), dataset2.getDatasetId(), now);
     Vote v2 = createFinalVote(dataset2.getCreateUserId(), e2.getElectionId());
 
-    Election e3 = createDataAccessElection(testDar1.getReferenceId(), dataset3.getDatasetId());
+    Election e3 = createDataAccessElection(testDar1.getReferenceId(), dataset3.getDatasetId(), now);
```

Comment:
Note that previously, this code was creating elections with different dates which implicitly means there were 3 separate times when a chairperson initiated the "Open" election operation. The application doesn't allow this behavior and enforces that a chairperson can only open elections for cases where there are no existing open elections.

When elections have the same creation date, that means the chairperson initiated a single "Open" election operation.

Reply:
I wonder if it might make sense to refactor `createDataAccessElection` to `createDataAccessElections` to a map of referenceId -> Set<datasetIds> so that this can be created in one transaction in the database.